server/src/ga/GaWorker.ts

```diff
@@ -31,55 +54,46 @@ export class GaWorker {
       new CounterTenorRule(),
     ];
 
-    console.log(`
-      Creating GaWorker with configs: 
-      iterations - ${GaWorker.ITERATIONS} , 
-      mutation_rate - ${GaWorker.MUTATION_RATE}
-    `);
+    console.log(`Creating GaWorker with config: ${JSON.stringify(config, null, 2)}`);
   }
 
-  /*
-      "A4,G3,F3,E3",
-      "F3,C4,B4,A4",
-      "C4,B4,A4,C4",
-   */
-
-  initialBots(randomInitial=false): Bot[] {
-
-    if(randomInitial){
-
-      let set = this.createStartSet();
-      let initialMelodies = randomInitialization(set, this.POPULATION_SIZE);
-      return initialMelodies.map((str) => new Bot(0, Melody.fromString(str)))
-
-    }else{
-
-      return [
+  initialBots(): Bot[] {
+    if (this.config.randomInitial) {
+      // Run GA once with randomly generated initial population
+      const set = this.createStartSet();
+      const initialMelodies = randomInitialization(set, this.config.selectionSize);
+      const bots = initialMelodies.map((melody) => new Bot(0, Melody.fromString(melody)));
+      return this.generateNewBots(bots);
+    } else {
+      // Return pregenerated melodies
+      const melodies = [
+        "A4,G3,F3,E3",
+        "F3,C4,B4,A4",
+        "C4,B4,A4,C4",
         "C4,G4,G4,F4",
         "A4,B4,G4,E4",
         "G4,B4,D5,E5",
         "E4,E4,E4,A4",
         "B4,A4,G4,D4",
         "D4,G4,C5,D5",
-        "E4,G4,E4,D4"
-      ].map((str) => new Bot(0, Melody.fromString(str)));
-
+        "E4,G4,E4,D4",
+      ];
+      const selectedMelodies = selectRandomMany(melodies, this.config.selectionSize);
+      return selectedMelodies.map((melody) => new Bot(0, Melody.fromString(melody)));
     }
-    
   }
 
   generateNewBots(startingPopulation: Bot[]): Bot[] {
     let generation = startingPopulation;
+    let fitnesses;
 
     // Number of generations to iterate before returning to client
-    for (let i = 0; i < GaWorker.ITERATIONS; i++) {
-      // Feature flagging
-
+    for (let i = 0; i < this.config.iterations; i++) {
       // Produce fitness scores from bots
-      let fitnesses = evaluate(generation);
+      fitnesses = evaluate(generation);
 
       // Normalize the scores to select a new generation
-      generation = selectRandomWeighted(generation, fitnesses, this.POPULATION_SIZE);
+      generation = selectRandomWeighted(generation, fitnesses, this.config.populationSize);
```

Comment:
Are we adding tests to make sure random weighted does what we want?
Also would be good to maybe have some hypothesis that this is actually what we want. Something related fitness score similar to some of our existing ones.

Reply:
Already have tests for it! Check out `Utils.test.ts`

Reply:
sweet!